File: go/1021.go
func removeOuterParentheses(S string) string {
	var c string
	counter := 0
	for _, v := range S {

		if v == '(' {
			if counter == 0 {

			} else {
				c = c + string(v)
			}
			counter++
		} else if v == ')' {
			if counter == 1 {

			} else {
				c = c + string(v)
			}
			counter--
		}

	}
	return c
}

File: java/1021.java
class Solution {
    public String removeOuterParentheses(String S) {
        int counter = 0;
        StringBuilder s = new StringBuilder();
        for(int i = 0;i<S.length();i++){
            if(S.charAt(i) == '('){
                if(counter != 0){
                    s.append(S.charAt(i));
                }
                counter ++;
            }
            else if(S.charAt(i) == ')'){
                if(counter > 1){
                    s.append(S.charAt(i));
                }
                counter --;
            }
        }
        return s.toString();
    }
}